### ISSUE TITLE
per_example_loss = -tf.reduce_sum(one_hot_labels * log_probs, axis=-1) 可否解释一下

### ISSUE THREAD

**Issue by binggoml** [huwenxianglyy/bert-use-demo#1]
你好楼主，在demo.py中，lost函数有一段代码：
    # probabilities = tf.nn.softmax(logits, axis=-1)
    log_probs = tf.nn.log_softmax(logits, axis=-1)
    one_hot_labels = tf.one_hot(input_labels, depth=num_labels, dtype=tf.float32)
    per_example_loss = -tf.reduce_sum(one_hot_labels * log_probs, axis=-1)
    loss = tf.reduce_mean(per_example_loss)

    predictions = tf.reshape(tf.argmax(logits, axis=-1, output_type=tf.int32),shape=[-1,1],name="predictions")
    acc = tf.reduce_mean(tf.cast(tf.equal(input_labels, predictions), "float"),
                         name="accuracy")

跟踪代码：
log_probs 的shape=（20,20） 第一个20表示batch，第二个20表示标签
one_hot_labels 的shape=(20, 1 , 20) 第一个20表示batch，第2个20表示标签，这个1不太清楚？
one_hot_labels * log_probs 的shape=(20, 20 , 20)
per_example_loss 为什么这么计算？
希望楼主能回答，万分感谢！

**Comment by huwenxianglyy** [comment]
这里是个失误，num_labels是类别的数量，我的代码中是个2分类。在参数中应该有个batch_size没有设置，之前一直把num_labels当做batch_size使用了。现在代码重新改过了
